data-structures/HeaderbasedCircularDoublyLinkedList.cpp

```diff
@@ -0,0 +1,216 @@
+#include <iostream>
+
+using namespace std;
+
+struct HCDNode {
```

Comment:
Please use this naming conventions https://google.github.io/styleguide/cppguide.html#Type_Names